server/src/main/java/org/apache/druid/server/coordinator/helper/DruidCoordinatorSegmentInfoLoader.java

```diff
@@ -42,8 +43,26 @@ public DruidCoordinatorRuntimeParams run(DruidCoordinatorRuntimeParams params)
   {
     log.info("Starting coordination. Getting available segments.");
 
-    // Display info about all available segments
-    final Set<DataSegment> availableSegments = coordinator.getOrderedAvailableDataSegments();
+    // The following transform() call doesn't actually transform the iterable. It only checks the sizes of the segments
+    // and emits alerts if segments with negative sizes are encountered. In other words, semantically it's similar to
+    // Stream.peek(). It works as long as DruidCoordinatorRuntimeParams.createAvailableSegmentsSet() (which is called
+    // below) guarantees to go over the passed iterable exactly once.
+    //noinspection StaticPseudoFunctionalStyleMethod: https://youtrack.jetbrains.com/issue/IDEA-153047
+    Iterable<DataSegment> availableSegmentsWithSizeChecking = Iterables.transform(
+        coordinator.iterateAvailableDataSegments(),
+        segment -> {
+          if (segment.getSize() < 0) {
+            log.makeAlert("No size on a segment")
```

Comment:
Could you reference the root cause of the no-size-segment?

Reply:
I don't know what is the root cause.